server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1813,6 +1831,41 @@ export class AgenticChatController implements ChatHandlers {
         return chatEventParser.getResult()
     }
 
+    /**
+     * This is needed to handle the case where a toolUseEvent takes a long time to resolve the stream and looks like
+     * nothing is happening.
+     */
+    async #showLoadingIfRequired(
```

Comment:
I appreciate that you isolated this logic! If I understand correctly, we will hopefully be able to remove this in the future once the backend resolves tool use faster.

Reply:
That's right, I think if we start seeing that the timeout starts resolving itself then we can remove this